test/test-connector-swagger-client.js

```diff
@@ -0,0 +1,74 @@
+// Copyright IBM Corp. 2016,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var should = require('should');
+var loopback = require('loopback');
+var http = require('http');
+
+function getMySwaggerClient() {
+  return {
+    execute: function(requestObject) {
+      if (requestObject.method === 'GET') {
```

Comment:
TBH, this custom swagger client looks unnecessary complicated to me. Since the purpose of the test is to verify that a custom client is used, it should be enough to write a stub client that always returns the same response:

```js
const customClient = {
  execute: requestObject => requestObject.on.response({id: 7});
};
```